ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala

```diff
@@ -281,13 +286,16 @@ private[state] object Conversions {
     )
 
   def decodeContractInstance(
-      coinst: TransactionOuterClass.ContractInstance
-  ): Value.VersionedContractInstance =
+      rawContractInstance: Raw.ContractInstance
+  ): Value.VersionedContractInstance = {
+    val contractInstance =
+      TransactionOuterClass.ContractInstance.parseFrom(rawContractInstance.bytes)
     assertDecode(
       "ContractInstance",
       TransactionCoder
-        .decodeVersionedContractInstance(ValueCoder.CidDecoder, coinst),
+        .decodeVersionedContractInstance(ValueCoder.CidDecoder, contractInstance),
     )
+  }
 
   def encodeContractInstance(
```

Comment:
Ideally you want to change the output type of `encodeContractInstance` to `Raw.ContractInstance`

Reply:
in order to have the encode symetric to the decode.